task.go

```diff
@@ -1,8 +1,65 @@
 package metre
 
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"time"
+)
+
 type Task struct {
-    ID string // Type Type of task (user as class prefix in cache)
-    Interval string // Schedule String in cron notation
-    Schedule func(t TaskRecord, s Scheduler, c Cache, q Queue)
-    Process func(t TaskRecord, s Scheduler, c Cache, q Queue)
+	MessageChannel chan string
+	StartTime      time.Time
+	TimeOut        time.Duration
+	MessageCount   int
+	ScheduleCount  int
+	ScheduleDone   bool
+	ID             string // Type Type of task (user as class prefix in cache)
+	Interval       string // Schedule String in cron notation
+	Schedule       func(t TaskRecord, s Scheduler, c Cache, q Queue)
+	Process        func(t TaskRecord, s Scheduler, c Cache, q Queue)
+}
+
+func (t Task) GetID() string {
+	return t.ID
+}
+
+func (t Task) GetInterval() string {
+	return t.Interval
+}
+
+func (t *Task) checkComplete() bool {
```

Comment:
This can be shortened to `return t.MessageCount == t.ScheduleCount && t.ScheduleDone`